### PR TITLE
core(artifacts): unify AnchorElements into single gatherer

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Unifies all our usage of `<a>` elements under a single `AnchorElements` gatherer/artifact. Replaces the two other artifacts (AnchorsWithNoRelOpener and CrawableLinks)

**Related Issues/PRs**
#6747

### CODE REVIEW

File: lighthouse-core/gather/gatherers/anchor-elements.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    catch (_) { return ''; }
+  };
+
+  /** @type {Array<HTMLAnchorElement|SVGAElement>} */
+  // @ts-ignore - put into scope via stringification
+  const anchorElements = getElementsInDocument('a');
+
+  return anchorElements.map(node => {
+    /** @type {HTMLAnchorElement} */
+    const anchorNode = (node);
```
[the remark this answers is not in the document]
Comment (brendankenny) [reply lighthouse-core/gather/gatherers/anchor-elements.js]: wdyt about something like
```js
return anchorElements.map(node => {
  // @ts-ignore - put into scope via stringification
  const outerHTML = getOuterHTMLSnippet(node); // eslint-disable-line no-undef

  if (node instanceof HTMLAnchorElement) {
    return {
      href: node.href,
      text: node.innerText,
      rel: node.rel,
      target: node.target,
      outerHTML,
    };
  }

  return {
    href: resolveURLOrEmpty(node.href.baseVal),
    text: node.textContent || '',
    rel: '',
    target: node.target.baseVal || '',
    outerHTML,
  };
});
```

(as far as I can tell, `SVGAElement.href` is always a `SVGAnimatedString`, see [SVGAElement](https://chromium.googlesource.com/chromium/blink/+/master/Source/core/svg/SVGAElement.idl) implementing [SVGURIReference](https://chromium.googlesource.com/chromium/blink/+/master/Source/core/svg/SVGURIReference.idl))

File: lighthouse-core/gather/gatherers/seo/crawlable-links.js [hunk 2]
```diff
@@ -1,44 +0,0 @@
...
-      const selector = 'a[href]:not([rel~="nofollow"])';
-      const elements = getElementsInDocument(selector);
-      return elements
-        .map(node => ({
-          href: node.href instanceof SVGAnimatedString ?
-            resolveURLOrNull(node.href.baseVal) :
-            node.href,
-          text: node.href instanceof SVGAnimatedString ?
-            node.textContent :
-            node.innerText,
```
Comment (brendankenny) [inline lighthouse-core/gather/gatherers/seo/crawlable-links.js]: also, do you know why the original code is `innerText ` for `HTMLAnchorElement` vs `textContent` for `SVGAElement`? We should keep it, just seems weird :)

Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/seo/crawlable-links.js]: I think we didn't want to return invisible text, IIRC there were some situations where we started sticking large amounts of nonsense text in the table

File: lighthouse-core/gather/gatherers/anchor-elements.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/anchor-elements.js]: yeah I like it!

**Pull request merged**